src/Symfony/Component/Translation/Translator.php

```diff
@@ -72,32 +72,45 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     private $configCacheFactory;
 
     /**
+     * @var TranslatorBagInterface
+     */
+    private $translatorBag;
+
+    /**
      * Constructor.
      *
-     * @param string               $locale   The locale
-     * @param MessageSelector|null $selector The message selector for pluralization
-     * @param string|null          $cacheDir The directory to use for the cache
-     * @param bool                 $debug    Use cache in debug mode ?
+     * @param string                 $locale        The locale
+     * @param MessageSelector|null   $selector      The message selector for pluralization
+     * @param string|null            $cacheDir      The directory to use for the cache
+     *                                              Deprecated since version 2.8, to be removed in 3.0.
+     * @param bool                   $debug         Use cache in debug mode ?
+     *                                              Deprecated since version 2.8, to be removed in 3.0.
+     * @param TranslatorBagInterface $translatorBag
      *
      * @throws \InvalidArgumentException If a locale contains invalid characters
      *
      * @api
      */
-    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false)
+    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false, TranslatorBagInterface $translatorBag = null)
```

Comment:
I wonder how BC break will be handled for this constructor.
If deprecations are removed for Symfony 3.0, then the constructor will become something like `__construct($locale, $selector, $translatorBag)`… which is a BC break.

What about adding a new Translator class and making this constructor deprecated? Plus the `translatorBag` param should always be required.

For instance:

```php
// Replace "Default" by a more apropriate name
class DefaultTranslator extends Translator
{
    public function __construct($locale, TranslatorBagInterface $translatorBag, MessageSelector $selector = null)
    {
         $this->setLocale($locale);
         $this->selector = $selector ?: new MessageSelector();
         $this->translatorBag = $translatorBag;
    }
}
```

This could also work for your other PR #15068